be/src/olap/memtable.cpp

```diff
@@ -425,6 +443,7 @@ Status MemTable::flush() {
     // and use the ids to load segment data file for calc delete bitmap.
     int64_t atomic_num_segments_before_flush = _rowset_writer->get_atomic_num_segment();
     RETURN_IF_ERROR(_do_flush(duration_ns));
+    _delta_writer_callback(_merged_rows);
```

Comment:
Why do we need this callback?

AFAIK, `_merged_rows` is already increased in 

```cpp
Status DeltaWriter::_flush_memtable_async() {
    _merged_rows += _mem_table->merged_rows();
    return _flush_token->submit(std::move(_mem_table));
}
```

The callstack (one of the code paths):

```cpp
DeltaWriter::_flush_memtable_async();
=> FlushToken::submit();
=> FlushToken::_flush_memtable();
=> new MemtableFlushTask();
=> MemtableFlushTask::run();
=> MemTable::flush(); // in the latest master, equivalent to MemTable::need_flush()
```

Reply:
OK I got it, `_mem_table->merged_rows()` was updated in `MemTable::_do_flush()`.

```cpp
    RETURN_IF_ERROR(_do_flush(duration_ns));
    delta_writer_callback(_merged_rows);
```

The question is, will it be added twice?

```cpp
// in DeltaWriter::_flush_memtable_async()
    _merged_rows += _mem_table->merged_rows(); // what if merged_rows() is non-zero here?

// in MemTable::flush() or MemTable::need_flush()
    _delta_writer_callback(_merged_rows)
```

I wonder if there is a better way to make this more clear to read.

Reply:
Oh yes, I think we should remove this line in _flush_memtable_async.
Now we only use _merged_rows to check correctness, and we can't get correct _merged_rows until flush has finished.